internal/gps/source_manager.go

```diff
@@ -188,11 +188,24 @@ func NewSourceManager(cachedir string) (*SourceMgr, error) {
 	// If it's a TemporaryError, we retry every second. Otherwise, we fail
 	// permanently.
 	//
-	// TODO: After some time, we should emit some kind of warning that we're waiting
-	// for the lockfile to be released. #534 should be address before we will do that.
+	// TODO: #534 needs to be implemented to provide a better way to log warnings,
+	// but until then we will just use stderr.
+	//
+	// #947 appears to be caused by some locking issue; as this is the most likely
+	// culprit, we will add some warnings here, so every 15 seconds we will emit
+	// a warning to stderr that we're waiting for the lockfile.
 
+	lasttime := time.Now()
 	err = lockfile.TryLock()
 	for err != nil {
+		nowtime := time.Now()
+		duration := nowtime.Sub(lasttime)
+
+		if duration > 15*time.Second {
```

Comment:
seems like this will only print after each 15 second window, rather than printing once initially and then again once every 15 seconds, no?